one-ui/src/main/java/com/marklogic/hub/oneui/auth/LoginLogoutHandler.java

```diff
@@ -43,19 +42,34 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
         ConnectionAuthenticationToken authenticationToken = (ConnectionAuthenticationToken)authentication;
         ObjectNode resp = mapper.createObjectNode();
         boolean isInstalled = authenticationToken.stagingIsAccessible();
+        final boolean[] managePrivilege = new boolean[3];
         if (authenticationToken.stagingIsAccessible()) {
             try {
                 RolesService rolesService = RolesService.on(authenticationToken.getHubConfigSession().newStagingClient());
                 resp.putArray("roles").addAll((ArrayNode) rolesService.getRoles());
+                if (authenticationToken.hasManagePrivileges()) {
```

Comment:
Pinging @ryanjdew on this - let's move this logic into ML - I don't know if we're referencing ML role names in our Java code yet, but I'd rather see our Java code at a more abstract level. 

In this case, the Java code needs to know if the user can install/upload. So let's tweak the role service to return both the set of roles (though I'm wondering what those are used for? I don't think the UI should be tightly bound to the names of ML roles) and e.g. "canInstall:true/false". This logic will then go into the roles endpoint, where we can easily unit test it.

Reply:
@rjrudin I agree, we should move this logic to ML. The trouble is that we cover the scenario where the Data Hub isn't installed yet, so hitting a Data Service isn't an option at that point.

It is something to revisit, but I think it will take more time to come to the right approach and I don't feel it should hold up this request.

Reply:
Ah, good point.